### PR TITLE
New build step macros warn

### LINKED ISSUE #629: New build step that checks for things that we might get caught up on
Grep for "TARGET_OS" in swift files and return an error since that means we're seeing ` #if TARGET_OS` and that won't work in Swift, but will still compile 🤦‍♂️

### PR DESCRIPTION
**Pull request by Juanpe**
Resolves #629

This PR aims to create a new build phase that runs a script to validate that some macros, like `TARGET_OS_ (TARGET_OS_IPHONE, TARGET_OS_SIMULATOR)`, are not included in any Swift file.

The script checks all the swift files into the `PurchasesCoreSwift` folder to find the specified regex.
If the script found some files, the project will not compile, and the developer can see this error:

Screenshot 2021-07-14 at 21 30 52

Screenshot 2021-07-14 at 21 31 44

### CODE REVIEW

File: Purchases.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -163,8 +163,6 @@
 		37E35B1F34D1624509012749 /* MockSubscriberAttributesManager.swift in Sources */ = {isa = PBXBuildFile; fileRef = 37E351D48260D9DC8B1EE360 /* MockSubscriberAttributesManager.swift */; };
 		37E35B4BDE7D7B1EFB3EE800 /* NSError+RCExtensions.m in Sources */ = {isa = PBXBuildFile; fileRef = 37E35088AB7C4F7A242D3ED5 /* NSError+RCExtensions.m */; };
 		37E35B94F66AD4C3BC3E3D54 /* RCPurchaserInfoManager.h in Headers */ = {isa = PBXBuildFile; fileRef = 37E35A375CFEFEE4D6DEFD11 /* RCPurchaserInfoManager.h */; settings = {ATTRIBUTES = (Private, ); }; };
-		37E35BDDC331C3A5FF72CFFF /* RCStoreKitRequestFetcher.m in Sources */ = {isa = PBXBuildFile; fileRef = 37E351CDC94FE0A49A3DAAB3 /* RCStoreKitRequestFetcher.m */; };
```
Comment (Juanpe) [inline Purchases.xcodeproj/project.pbxproj:166]: I'm not sure why these changes are included here 🤔

File: Purchases.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1654,6 +1651,24 @@
...
+			inputPaths = (
+			);
+			name = "check TARGET_OS macro";
+			outputFileListPaths = (
+			);
+			outputPaths = (
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = /bin/sh;
+			shellScript = "#!/usr/bin/env bash\n\nif grep -l \"TARGET_OS\" $(find \"${PROJECT_DIR}/PurchasesCoreSwift\" -name \"*.swift\")\nthen\n  echo \"error: TARGET_OS macro has been found in some files\"\n  exit 1\nfi\n";
```
Comment (taquitos) [inline Purchases.xcodeproj/project.pbxproj:1670]: We're likely going to rename `PurchasesCoreSwift` eventually, so maybe we can just search through the entire project folder (including the `Purchases`) folder for swift files. That way, if we change folder structure and forget about this, it will still catch things.

File: Purchases.xcodeproj/project.pbxproj (hunk 1 above)
Comment (taquitos) [reply Purchases.xcodeproj/project.pbxproj:166]: Hah, classic Xcode. Could you exclude these from the change set?

**Review by taquitos — CHANGES_REQUESTED** [review]
This looks 💯 
A couple tiny changes, but then it's :shipit:  time

**Comment by Juanpe** [comment]
I've a problem with the xcodeproj file, because of that I'm going to close this PR and create a new one
